iOS/Flows/Channel/Manager/ChannelSupplier.swift

```diff
@@ -178,12 +178,17 @@ class ChannelSupplier {
 
     // MARK: GETTERS
 
-    func waitForInitialSync() -> Future<Void, Never> {
-        return Future { promise in
-            ChannelSupplier.shared.$isSynced.mainSink { (isSynced) in
-                guard isSynced else { return }
-                promise(.success(()))
-            }.store(in: &self.cancellables)
+    func waitForInitialSync() async {
+        var cancellable: AnyCancellable?
+
+        return await withCheckedContinuation { continuation in
+            cancellable = ChannelSupplier.shared.$isSynced
+                .mainSink { (isSynced) in
+                    guard isSynced else { return }
+
+                    continuation.resume(returning: ())
+                    cancellable?.cancel()
+                }
```

Comment:
A little tricky to figure out. I make a local var, "cancellable", so I can hold on to the subscriber in memory. No need to assign to our member array of cancellable because the close captures the variable and keeps it alive.